src/sage/groups/braid.py

```diff
@@ -1658,6 +1750,91 @@ def is_conjugated(self, other):
         l = conjugatingbraid(self, other)
         return bool(l)
 
+    def pure_conjugating_braid(self, other):
+        r"""
+        Return a pure conjugating braid, i.e. a conjugating braid whose
+        associated permutation is the identity, if it exists.
+
+        INPUT:
+
+        - ``other`` -- a braid in the same braid group as ``self``
+
+        OUTPUT:
+
+        A pure braid `d` such that if `o` equals ``other`` and `s` equals ``self``
+        then `o = d^{-1} \cdot s \cdot d`.
+
+        EXAMPLES::
+
+            sage: B = BraidGroup(4)
+            sage: S = SymmetricGroup(4)
+            sage: a = B([1, 2, 3])
+            sage: b = B([3, 2,])
+            sage: c = b ^ 12 * a / b ^ 12
+            sage: d1 = a.conjugating_braid(c)
+            sage: print(len(d1.Tietze()))
+            30
+            sage: print(d1.permutation(W=S))
+            (1,3)(2,4)
+            sage: d1 * c / d1 == a
+            True
+            sage: d1 * a / d1 == c
+            False
+            sage: d2 = a.pure_conjugating_braid(c)
+            sage: print(len(d2.Tietze()))
```

Comment:
You don't need all these `print`s here.

Reply:
Done for last suggestions, I will upload the changes soon